src/uptane/secondaryconfig.h

```diff
@@ -1,17 +1,23 @@
 #ifndef UPTANE_SECONDARYCONFIG_H_
 #define UPTANE_SECONDARYCONFIG_H_
 
+#include <string>
+
 #include <boost/filesystem.hpp>
 
-namespace Uptane{
-struct SecondaryConfig{
-    std::string ecu_serial;
-    std::string ecu_hardware_id;
-    std::string ecu_private_key;
-    std::string ecu_public_key;
-    boost::filesystem::path full_client_dir;
-    bool partial_verifying;
-    boost::filesystem::path firmware_path;
+namespace Uptane {
+class SecondaryConfig {
+ public:
+  std::string ecu_private_key() const;
+  std::string ecu_public_key() const;
+
+  std::string ecu_serial;
+  std::string ecu_hardware_id;
+  std::string ecu_private_key_;
```

Comment:
ecu_private_key_ and ecu_public_key_ should be private?

Reply:
I have wondered about that. For now, I'm following the convention we set with the TlsConfig, which is that you can get/set the strings directly, but if you want to get the path with the directory prefixed automatically, you have to use the function. We could perhaps use a different name for the function if that would help make it clearer.